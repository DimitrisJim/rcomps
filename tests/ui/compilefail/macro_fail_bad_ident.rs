// Check cases that compilefail.
//
// See https://rustc-dev-guide.rust-lang.org/tests/adding.html#error-annotations
// for annotating the snippets with the expected error.
// ignore-windows
extern crate rcomps;
use rcomps::comp;

/// Bad identifier checks
fn test_fail_bad_ident(){
    let c = || {42};
    comp!([for 40 in 1..2 => 30]);     //~ ERROR Unable to parse expression.
    comp!({for 40 in 1..2 => 0});      //~ ERROR Unable to parse expression.
    comp!({for 40 in 1..2 => 0, 0});   //~ ERROR Unable to parse expression.

    // Second error concerns mismatched types. TODO: Why is it raised?
    comp!([for (2 > 1) in 1..2 => 20]);     //~ ERROR Unable to parse expression.
    comp!({for (2 > 1) in 1..2 => 20});     //~ ERROR Unable to parse expression.
    comp!({for (2 > 1) in 1..2 => 20, 20}); //~ ERROR Unable to parse expression.

    comp!([for c() in 1..2 => 0]);       //~ ERROR Unable to parse expression.
    comp!([for (c()) in 1..2 => 0]);     //~ ERROR Unable to parse expression.
    comp!({for c() in 1..2 => 0});       //~ ERROR Unable to parse expression.
    comp!({for (c()) in 1..2 => 0});     //~ ERROR Unable to parse expression.
    comp!({for c() in 1..2 => 0, 0});    //~ ERROR Unable to parse expression.
    comp!({for (c()) in 1..2 => 0, 0});  //~ ERROR Unable to parse expression.

    comp!([for r#self in 3..6 => 0]);       //~ ERROR `self` cannot be a raw identifier
    comp!({for r#self in 3..6 => 0});       //~ ERROR `self` cannot be a raw identifier
    comp!({for r#self in 3..6 => 0, 0});    //~ ERROR `self` cannot be a raw identifier
}

fn main(){}