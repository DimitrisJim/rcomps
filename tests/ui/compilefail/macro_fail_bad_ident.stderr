error: `self` cannot be a raw identifier
  --> $DIR/macro_fail_bad_ident.rs:28:16
   |
28 |     comp!([for r#self in 3..6 => 0]);       //~ ERROR `self` cannot be a raw identifier
   |                ^^^^^^

error: `self` cannot be a raw identifier
  --> $DIR/macro_fail_bad_ident.rs:29:16
   |
29 |     comp!({for r#self in 3..6 => 0});       //~ ERROR `self` cannot be a raw identifier
   |                ^^^^^^

error: `self` cannot be a raw identifier
  --> $DIR/macro_fail_bad_ident.rs:30:16
   |
30 |     comp!({for r#self in 3..6 => 0, 0});    //~ ERROR `self` cannot be a raw identifier
   |                ^^^^^^

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:12:5
   |
12 |     comp!([for 40 in 1..2 => 30]);     //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:13:5
   |
13 |     comp!({for 40 in 1..2 => 0});      //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:14:5
   |
14 |     comp!({for 40 in 1..2 => 0, 0});   //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:17:5
   |
17 |     comp!([for (2 > 1) in 1..2 => 20]);     //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:18:5
   |
18 |     comp!({for (2 > 1) in 1..2 => 20});     //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:19:5
   |
19 |     comp!({for (2 > 1) in 1..2 => 20, 20}); //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:21:5
   |
21 |     comp!([for c() in 1..2 => 0]);       //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:22:5
   |
22 |     comp!([for (c()) in 1..2 => 0]);     //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:23:5
   |
23 |     comp!({for c() in 1..2 => 0});       //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:24:5
   |
24 |     comp!({for (c()) in 1..2 => 0});     //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:25:5
   |
25 |     comp!({for c() in 1..2 => 0, 0});    //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unable to parse expression.
  --> $DIR/macro_fail_bad_ident.rs:26:5
   |
26 |     comp!({for (c()) in 1..2 => 0, 0});  //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 15 previous errors

